marklogic-data-hub/src/main/java/com/marklogic/hub/dhs/DhsDeployer.java

```diff
@@ -55,6 +57,26 @@ public void deployAsSecurityAdmin(HubConfigImpl hubConfig) {
         dhsDeployer.deploy(hubConfig.getAppConfig());
     }
 
+    /**
+     * Compares current ML version with the minimum required version and fails immediately
+     * if current ML version is lower than the minimum required version.
+     * @param hubConfig
+     */
+    public void checkCompatibility(HubConfig hubConfig) {
+        Versions versions = new Versions(hubConfig);
+        String mlVersionString = versions.getMarkLogicVersion();
+        Versions.MarkLogicVersion mlVersion = versions.getMLVersion(mlVersionString);
+        if (!((mlVersion.getMajor() > 10) || (mlVersion.getMajor() == 10 && mlVersion.getMinor() >= 201) || (mlVersion.getMajor() == 9 && mlVersion.getMinor() >= 1100))) {
+            logger.warn("Cannot proceed as this version of Data Hub does not support the detected version of MarkLogic:");
```

Comment:
Per the JIRA ticket, let's make this the actual error message. The problem with logging is that the user will see the error message below and may see a stacktrace as well, but this logging will be above all of that. So the user may miss it. Let's make it obvious and build up a string as described in the JIRA ticket, and then use that as the exception message.